pytorch_lightning/metrics/metric.py

```diff
@@ -333,7 +333,11 @@ def __hash__(self):
         hash_vals = [self.__class__.__name__]
 
         for key in self._defaults.keys():
-            hash_vals.append(getattr(self, key))
+            val = getattr(self, key)
+            # Special case: allow list values, so long as their elements are hashable
+            if isinstance(val, list):
```

Comment:
are lists the only issue here? should we do it with sequences instead?

Reply:
This is a good point. I was erring on the side of caution, but perhaps it was too cautious. What about this condition instead?
```
if hasattr(val, '__iter__') and not isinstance(val, torch.Tensor):
```